kw_webapp/models.py

```diff
@@ -271,15 +273,19 @@ class UserSpecific(models.Model):
     class Meta:
         unique_together = ("vocabulary", "user")
 
-    def answered_correctly(self, first_try=True):
+    def answered_correctly(self, first_try, can_burn):
         # This is a check to see if it is a "lesson" object.
         if self.streak == 0:
             self.streak += 1
         elif first_try:
             self.correct += 1
             self.streak += 1
             if self.streak >= constants.WANIKANI_SRS_LEVELS[WkSrsLevel.BURNED.name][0]:
-                self.burned = True
+                # If can burn, do so. Otherwise, keep review at enlightened.
+                if can_burn:
+                    self.burned = True
+                else:
+                    self.streak = constants.WANIKANI_SRS_LEVELS[WkSrsLevel.ENLIGHTENED.name][0]
```

Comment:
Took a quick look to see if there were any changes needed for the review queue, but since a new review time is generated and the review is kept at enlightened, I think things should just work.